app/src/main/java/com/esri/arcgisruntime/opensourceapps/datacollection/views/popup/PopupView.kt

```diff
@@ -132,22 +338,97 @@ class PopupView : FrameLayout {
             oldItem: PopupAttribute,
             newItem: PopupAttribute
         ): Boolean {
-            return oldItem.field == newItem.field && oldItem.value == newItem.value
+            return oldItem.fieldName == newItem.fieldName && oldItem.value == newItem.value
         }
     }
 
     /**
      * The PopupAttributeListAdapter ViewHolder.
      */
-    private class ViewHolder(private val binding: ViewDataBinding) :
+    private inner class ViewHolder(private val binding: ViewDataBinding) :
         RecyclerView.ViewHolder(binding.root) {
 
+        val popupFieldValue: TextView by lazy {
+            binding.root.popupFieldValue
+        }
+
+        val popupFieldEditValue: EditText by lazy {
+            binding.root.popupFieldEditValue
+        }
+
         fun bind(
             popupAttribute: PopupAttribute
         ) {
             binding.setVariable(BR.popupAttribute, popupAttribute)
             binding.executePendingBindings()
         }
+
+        /**
+         * Toggles the view for popup field value from edittext to textview and vice-versa, given the
+         * edit mode of the popupView.
+         */
+        fun setViewsBehavior(popupAttribute: PopupAttribute) {
+            if (isEditable(popupAttribute)) {
+                popupFieldEditValue.inputType = getInputType(popupAttribute.fieldType)
+                popupFieldEditValue.visibility = View.VISIBLE
+                popupFieldValue.visibility = View.GONE
+                // here we assign and hold the values of the editable fields, entered by the user
+                // in popupAttribute.tempValue
+                popupFieldEditValue.doAfterTextChanged {
+                    popupAttribute.tempValue = popupFieldEditValue.text}
+            } else {
+                popupFieldEditValue.visibility = View.GONE
+                popupFieldValue.visibility = View.VISIBLE
+                popupAttribute.tempValue = null
+            }
+        }
+
+        /**
+         * Returns whether the popup attribute is editable based on the status of popupView edit mode
+         * and if the popup attribute is editable.
+         */
+        fun isEditable(popupAttribute: PopupAttribute): Boolean {
+            return editMode && popupAttribute.isEditable
```

Comment:
This is just a style question, is there a kotlin preferred style for member variable names? I was expecting something like `mEditMode` when I was looking for the reasoning of making this class `inner` and I noticed that none of the members of PopupView use this style.

Reply:
the `mEditMode` is java style. For Kotlin see here
https://kotlinlang.org/docs/reference/coding-conventions.html#naming-rules